system/jlib/jfile.cpp

```diff
@@ -7947,38 +7933,63 @@ MODULE_INIT(INIT_PRIORITY_STANDARD)
-            if (plane.hasProp(propName.c_str()))
-                values[FileSyncWriteClose] = plane.getPropBool(propName.c_str()) ? 1 : 0;
-            else
+            for (unsigned propNum=0; propNum<PlaneAttributeType::PlaneAttributeCount; ++propNum)
             {
-                // temporary (check legacy fileSyncMaxRetrySecs too), purely for short term backward compatibility (see HPCC-xxxx)
-                unsigned __int64 v = plane.getPropInt64("expert/@fileSyncMaxRetrySecs", unsetPlaneAttrValue);
-                // NB: fileSyncMaxRetrySecs==0 is treated as set/enabled
-                values[FileSyncWriteClose] = v != unsetPlaneAttrValue ? 1 : 0;
+                const PlaneAttributeInfo &attrInfo = planeAttributeInfo[propNum];
+                std::string prop;
+                if (attrInfo.isExpert)
+                    prop += "expert/";
+                prop += "@" + std::string(attrInfo.name);
+                switch (attrInfo.type)
+                {
+                    case PlaneAttrType::integer:
+                    case PlaneAttrType::boolean: // handling is same as integer
+                    {
+                        unsigned __int64 value = plane.getPropInt64(prop.c_str(), unsetPlaneAttrValue);
+                        if (unsetPlaneAttrValue != value)
+                        {
+                            if (attrInfo.scale)
+                            {
+                                dbgassertex(PlaneAttrType::integer == attrInfo.type);
+                                value *= attrInfo.scale;
+                            }
+                        }
+                        else if (FileSyncWriteClose == propNum) // temporary (if FileSyncWriteClose and unset, check legacy fileSyncMaxRetrySecs), purely for short term backward compatibility (see HPCC-32757)
+                        {
+                            unsigned __int64 v = plane.getPropInt64("expert/@fileSyncMaxRetrySecs", unsetPlaneAttrValue);
+                            // NB: fileSyncMaxRetrySecs==0 is treated as set/enabled
+                            value = v != unsetPlaneAttrValue ? 1 : 0;
+                        }
+                        values[propNum] = value;
+                        break;
+                    }
+                    default:
+                        throwUnexpected();
+                }
             }
         }
 
         // reset defaults
         expertEnableIFileFlagsMask = IFEnone;
         expertDisableIFileFlagsMask = IFEnone;
 
+        Owned<IPropertyTree> componentConfig = getComponentConfig();
         StringBuffer fileFlagsStr;
-        if (getComponentConfigSP()->getProp("expert/@enableIFileMask", fileFlagsStr) || getGlobalConfigSP()->getProp("expert/@enableIFileMask", fileFlagsStr))
+        if (componentConfig->getProp("expert/@enableIFileMask", fileFlagsStr) || getGlobalConfigSP()->getProp("expert/@enableIFileMask", fileFlagsStr))
```

Comment:
minor: Could also move getGlobalConfigSP() into  local to avoid multiple calls.

Reply:
will change